inference-engine/src/transformations/src/transformations/op_conversions/convert_sequences_to_tensor_iterator.cpp

```diff
@@ -61,101 +57,140 @@ namespace {
+        std::shared_ptr<ngraph::Node> reverse_seq_before;
         if (is_reverse && enable_mask) {
-            reverse_seq_before = std::make_shared<opset5::ReverseSequence>(X, seq_lengths, 0, 1);
+            reverse_seq_before = std::make_shared<ngraph::opset5::ReverseSequence>(X, seq_lengths, 0, 1);
         }
 
         // TensorIterator Body: begin
-        Shape X_param_shape = X.get_shape();
-        X_param_shape.at(1) = 1; // split by seq_lengths dimension
-        auto X_body_param = std::make_shared<opset5::Parameter>(X.get_element_type(), X_param_shape);
-        auto H_body_param = std::make_shared<opset5::Parameter>(H_t.get_element_type(),
-                                                                H_t.get_shape());
-        auto seq_body_param = std::make_shared<opset5::Parameter>(seq_lengths.get_element_type(),
-                                                                  seq_lengths.get_partial_shape());
-
-        auto axis_0 = std::make_shared<opset5::Constant>(element::i64, Shape{1}, std::vector<int64_t>{0});
-        auto axis_1 = std::make_shared<opset5::Constant>(element::i64, Shape{1}, std::vector<int64_t>{1});
-
-        const auto& ins = squeeze_nodes({X_body_param, H_body_param, W, R, B}, {axis_1, axis_1, axis_0, axis_0, axis_0});
-        auto cell = std::make_shared<opset5::RNNCell>(ins[0],
-                                                      ins[1],
-                                                      ins[2],
-                                                      ins[3],
-                                                      ins[4],
-                                                      sequence->get_hidden_size(),
-                                                      sequence->get_activations(),
-                                                      sequence->get_activations_alpha(),
-                                                      sequence->get_activations_beta(),
-                                                      sequence->get_clip());
-
-        ParameterVector body_params;
-        ResultVector body_results;
-        auto unsqueeze_dum_dir = std::make_shared<opset5::Unsqueeze>(cell->output(0), axis_1);
-        Output<Node> h_node_to_result = unsqueeze_dum_dir;
+        auto X_param_pshape = X_pshape;
+        X_param_pshape[1] = 1; // split by seq_lengths dimension
+        auto X_body_param = std::make_shared<ngraph::opset5::Parameter>(X.get_element_type(), X_param_pshape);
+        auto H_body_param = std::make_shared<ngraph::opset5::Parameter>(H_t.get_element_type(), H_t.get_partial_shape());
+        auto seq_body_param = std::make_shared<ngraph::opset5::Parameter>(seq_lengths.get_element_type(), seq_lengths.get_partial_shape());
+
+        // LSTM case
+        auto C_body_param = cell_state_defined ?
+            std::make_shared<ngraph::opset5::Parameter>(C_t.get_element_type(), C_t.get_partial_shape()) :
+            std::make_shared<ngraph::opset5::Parameter>();
```

Comment:
do you wanna create empty shared_ptr here?` std::make_shared<ngraph::opset5::Parameter>()`?

Reply:
We need this parameter only in case of LSTM sequence. Empty Parameter replaced with nullptr